processing/src/main/java/org/apache/druid/utils/CompressionUtils.java

```diff
@@ -657,7 +659,7 @@ public static OutputStream compress(final OutputStream in, final Format format)
       case XZ: return new XZCompressorOutputStream(in);
       case SNAPPY: return new FramedSnappyCompressorOutputStream(in);
       case ZSTD: return new ZstdCompressorOutputStream(in);
-      case ZIP: return new ZipOutputStream(in, StandardCharsets.UTF_8);
+      case ZIP: return new DeflaterOutputStream(in);
```

Comment:
The ZipOutputStream / ZipInputStream expect to take place on entries inside a ZipEntry. Since these helpers are intended for compressing/decompressing bytes directly, we instead use the DeflaterOutputStream / InflaterInputStream underlying compression classes for zip.

Reply:
On second thought, this doesn't make sense as a solution since zip isn't necessary deflate. 

An alternative would be to just use a dummy zipEntry with a single optimistic getNextEntry on decompress - or, perhaps, to just disable zip as a valid compression format for Kinesis since it doesn't make much sense in a record-streaming context.